zeth_tool/main.cpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2015-2021 Clearmatics Technologies Ltd
+//
+// SPDX-License-Identifier: LGPL-3.0+
+
+#include "libtool/run.hpp"
+#include "zeth_tool/dump_proof_cmd.hpp"
+#include "zeth_tool/prove_cmd.hpp"
+#include "zeth_tool/tool_common.hpp"
+#include "zeth_tool/verify_cmd.hpp"
+
+using namespace zethtool;
+
+class zeth_command : public libtool::command<global_options>
+{
+public:
+    void initialize_global_options(
+        boost::program_options::options_description &global,
+        boost::program_options::options_description &all_options) override
+    {
+        global.add_options()("verbose,v", "Verbose output");
+        all_options.add(global);
+    }
+
+    /// Parse the variables map to update the GlobalOptionsT object.
+    void parse_global_options(
+        global_options &out_options,
+        const boost::program_options::variables_map &vm) override
+    {
+        out_options = (bool)vm.count("verbose");
+    }
+};
+
+int main(int argc, char **argv)
+{
+    // Create command structures
+    std::map<std::string, zeth_subcommand *> commands{
+        {"verify", verify_cmd},
+        {"prove", prove_cmd},
+        {"dump-proof", dump_proof_cmd},
+    };
```

Comment:
If I try invalid command names, I get a segmentation faults e.g.
```bash
17:21:56 ~/D/t/z/build ❯❯❯ ./zeth_tool/zeth-tool verif                            ✘ 1
[1]    25886 segmentation fault (core dumped)  ./zeth_tool/zeth-tool verif
17:22:11 ~/D/t/z/build ❯❯❯ ./zeth_tool/zeth-tool generate                       ✘ 139
[1]    25935 segmentation fault (core dumped)  ./zeth_tool/zeth-tool generate
```

That would be preferable to fail safe here and simply display a message `command not found` and fallback to the help menu.

Reply:
fixed.  thanks.

Reply:
Great, thanks!